### DISCUSSION TITLE
May I know the pull request or the commit id for enabling the cpd off/on for Java?

### DISCUSSION THREAD

**Discussion by prashanthr84** [pmd/pmd#2550] [Q&A] [answered]
I would like to see How actually the cpd off/on to suppress the code duplication is implemented in java or may be in any other language. Commit id or pull request might be of help. Any specific reason for not implementing yet for C#?

**Comment by jsotuyod** [comment, marked as answer]
Actually, C# does support `CPD-OFF` / `CPD-ON` comment-based suppressions… C# is based on an Antlr grammar, and the base `AntlrTokenFilter` shipped within pmd-core offers this.

I guess the user docs are outdated…
